### PR TITLE
Document new features and write release announcement.

### LINKED ISSUE #598: Document new functionality
A lot has happened since the last stable release. Before releasing v0.5, we should update the website to document new functionality such as

- completions
- hover
- signature help

In particular, we should validate all the features work reliably in the supported editors (vim, emacs, sublime, atom).

### PR DESCRIPTION
**Pull request by olafurpg**
Fixes #598

### CODE REVIEW

File: docs/editors/overview.md [hunk 1]
```diff
@@ -146,18 +175,59 @@ buffer.
...
 
+## Completions
+
+**✅ Atom**: no auto-import and string interpolator completions produce invalid
+code.
+
+**✅ Vim**: auto-import and snippets require
+[coc.nvim](https://github.com/neoclide/coc.nvim) client.
+
+**✅Sublime**: no auto-import.
```
Comment (gabro) [inline docs/editors/overview.md]:
```suggestion
**✅ Sublime**: no auto-import.
```

File: docs/editors/overview.md [hunk 2]
```diff
@@ -146,18 +175,59 @@ buffer.
...
+
+- **Auto-import**: symbols that are not in scope are automatically imported
+  locally. Imports still need to be organized manually, we are exploring ways to
+  automate this workflow in the future.
+- **Override def**: implement methods from the super class.
+- **Exhaustive match**: generate an exhaustive pattern match for sealed types.
+- **String interpolator**: automatically convert string literals into string
+  interpolators.
+- **Filename**: complete classnames based on the enclosing file.
+- **Documentation**: read the docstring for for method symbols by pressing
```
Comment (gabro) [inline docs/editors/overview.md]:
```suggestion
- **Documentation**: read the docstring for method symbols by pressing
```

File: docs/editors/overview.md [hunk 3]
```diff
@@ -146,18 +175,59 @@ buffer.
...
+
+## Hover (aka. type at point)
+
+See the expression type and symbol signature under the cursor.
+
+![](https://i.imgur.com/2MfQvsM.gif)
+
+- **Expression type**: shows the non-generic type of the highlighted expression.
+- **Symbol signature**: shows the generic signature of symbol under the cursor
+  along with it's docstring, if available.
```
Comment (gabro) [inline docs/editors/overview.md]:
```suggestion
  along with its docstring, if available.
```

File: docs/editors/overview.md [hunk 4]
```diff
@@ -189,6 +259,7 @@ Fuzzy search a symbol in the workspace of library dependencies by its name.
 
 - All-lowercase queries are treated as case-insensitive searches.
 - Queries ending with a dot `.` list nested symbols.
+- Queries with a semicolon `;` search library dependencies.
```
Comment (gabro) [inline docs/editors/overview.md]:
```suggestion
- Queries containing a semicolon `;` search library dependencies.
```

File: docs/editors/vim.md [hunk 5]
```diff
@@ -3,8 +3,21 @@ id: vim
...
 
-Metals works in vim thanks to the the
-[`vim-lsc`](https://github.com/natebosch/vim-lsc/) extension.
+Metals works with most LSP clients for Vim:
+
+- [`vim-lsc`](https://github.com/natebosch/vim-lsc/): simple installation and
+  low resource usage but limited functionality (no auto-import, cancellation,
+  formatting, folding).
+- [`coc.nvim`](https://github.com/neoclide/coc.nvim): installation requires
+  neovim or vim v8.1 along with npm. Feature rich, supports all of LSP.
```
Comment (gabro) [inline docs/editors/vim.md]:
```suggestion
  neovim or Vim v8.1 along with npm. Feature rich, supports all of LSP.
```

File: test-workspace/src/main/scala/example/User.scala [hunk 6]
```diff
@@ -1,12 +1,8 @@
...
-  println(Option(1).map(_.toString()))
-  import java.nio.file.Files
-  import java.nio.file.Paths
-  List(1).map(identity _)
-  System.out.println(identity(), identity(), identity())
-  Files.readAllBytes(Paths.get(""))
+object Main {
+	"".su
+	// Files
+	// Files
```
Comment (gabro) [inline test-workspace/src/main/scala/example/User.scala]: intended?

File: website/blog/2020-04-12-mercury.md [hunk 7]
```diff
@@ -0,0 +1,331 @@
+---
+author: Ólafur Páll Geirsson
+title: Metals v0.5.0
+authorURL: https://twitter.com/olafurpg
+authorImageURL: https://avatars2.githubusercontent.com/u/1408093?s=460&v=4
+---
+
+We are excited to announce the release of Metals v0.5, codename "Mercury" 🎉
+This is release introduces new features, performance improvements and bug fixes
```
Comment (gabro) [inline website/blog/2020-04-12-mercury.md]:
```suggestion
This release introduces new features, performance improvements and bug fixes
```

File: website/blog/2020-04-12-mercury.md [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+
+- **Auto-import**: symbols that are not in scope are automatically imported
+  locally. Imports still need to be organized manually, we are exploring ways to
+  automate this workflow in the future.
+- **Override def**: implement methods from the super class.
+- **Exhaustive match**: generate an exhaustive pattern match for sealed types.
+- **String interpolator**: automatically convert string literals into string
+  interpolators.
+- **Filename**: complete classnames based on the enclosing file.
+- **Documentation**: read the docstring for for method symbols by pressing
```
Comment (gabro) [inline website/blog/2020-04-12-mercury.md]:
```suggestion
- **Documentation**: read the docstring for method symbols by pressing
```

File: website/blog/2020-04-12-mercury.md [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+## Hover (aka. type at point)
+
+It's now possible to see the expression type and symbol signature under the
+cursor.
+
+![](https://i.imgur.com/2MfQvsM.gif)
+
+- **Expression type**: shows the non-generic type of the highlighted expression.
+- **Symbol signature**: shows the generic signature of symbol under the cursor
+  along with it's docstring, if available.
```
Comment (gabro) [inline website/blog/2020-04-12-mercury.md]:
```suggestion
  along with its docstring, if available.
```

File: test-workspace/src/main/scala/example/User.scala (hunk 6 above)
Comment (olafurpg) [reply test-workspace/src/main/scala/example/User.scala]: Nope, will revert.

**Pull request merged**